src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs

```diff
@@ -465,7 +466,9 @@ internal async Task ExecuteTarget(ITaskBuilder taskBuilder, BuildRequestEntry re
 
                         // UNDONE: (Refactor) Refactor TargetUpToDateChecker to take a logging context, not a logging service.
                         TargetUpToDateChecker dependencyAnalyzer = new TargetUpToDateChecker(requestEntry.RequestConfiguration.Project, _target, targetLoggingContext.LoggingService, targetLoggingContext.BuildEventContext);
+                        MSBuildEventSource.Log.PerformDependencyAnalysisStart();
                         DependencyAnalysisResult dependencyResult = dependencyAnalyzer.PerformDependencyAnalysis(bucket, out changedTargetInputs, out upToDateTargetInputs);
+                        MSBuildEventSource.Log.PerformDependencyAnalysisStop();
```

Comment:
Can this log the result? I think we'll be interested in two aspects:

1. Time it takes to decide something's fully up to date (pure overhead in a no-op incremental build)
2. Time it takes to decide something's _not_ up to date in "obvious" cases (overhead when we need to build).

Reply:
It definitely can...I don't think it can log what the result really means, though. In other words, it can log the int corresponding to the enum, but I don't think it can log what "0" or "4" correspond to. I can add it in anyway, since you can always look it up if it's important to you.

Reply:
That's good enough for me too.